alibi/explainers/similarity/grad.py

```diff
@@ -0,0 +1,203 @@
+"""Gradient-based explainer.
+
+This module implements the gradient-based explainers grad-dot and grad-cos.
+"""
+
+import copy
+from typing import TYPE_CHECKING, Callable, Optional, Union, Dict, Tuple
+
+import numpy as np
+
+from alibi.api.interfaces import Explanation
+from alibi.explainers.similarity.base import BaseSimilarityExplainer
+from alibi.explainers.similarity.metrics import dot, cos, asym_dot
+from alibi.api.defaults import DEFAULT_META_SIM, DEFAULT_DATA_SIM
+
+if TYPE_CHECKING:
+    import tensorflow
+    import torch
+
+
+class SimilarityExplainer(BaseSimilarityExplainer):
+    def __init__(self,
+                 predictor: 'Union[tensorflow.keras.Model, torch.nn.Module]',
+                 loss_fn: '''Callable[[Union[tensorflow.Tensor, torch.Tensor],
+                                    Union[tensorflow.Tensor, torch.Tensor]],
+                                   Union[tensorflow.Tensor, torch.Tensor]]''',
```

Comment:
I think you should split it in two:
Union[Callable[[tf.Tensor, tf.tensor], tf.Tensor], Callable[[toch.Tensor, torch.Tensor], torch.Tensor]]

Otherwise, I think it would mean something else.

Reply:
Yup, make consistent with the base class. Also, are the triple quotes necessary?